File: 238-product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans(nums.size());
        ans[0] = 1;
        for (int i = 1; i < nums.size(); ++i){
            ans[i] = nums[i-1] * ans[i-1]; 
        }
        int rig = 1;
        for (int i = nums.size() - 1; i >= 0; --i){
            ans[i] = ans[i] * rig;
            rig *= nums[i]; 
        }
        return ans;
    }
};

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ans = [1] * len(nums)
        produs = 1
        for i in range (1, len(nums)):
            ans[i] = ans[i-1] * nums[i-1]
        for j in range (len(nums) - 1, -1, -1):
            ans[j] = ans[j] * produs
            produs *= nums[j]
        return ans
        